pyterrier/pipelines.py

```diff
@@ -506,6 +563,48 @@ def _apply_round(measure, value):
         'total' : len(retr_systems),
         'desc' : 'pt.Experiment'
     }
+    
+    
+    common_pipe, execution_retr_systems = _identifyCommon(retr_systems)
```

Comment:
This function is getting a bit unwieldy. Maybe it's worth splitting out this bit into a helper function?